Build failure with boost-1.61

On my gentoo amd64 machine with gcc-5.4.0 I see the following build failure with boost-1.61.

<pre>
/var/tmp/portage/games-strategy/freeorion-0.4.5/work/freeorion-0.4.5/parse/EffectParser5.cpp:27:17:   required from here
/usr/include/boost/spirit/home/qi/detail/alternative_function.hpp:30:46: error: no type named ‘types’ in ‘boost::spirit::qi::detail::find_substitute<boost::optional<boost::variant<std::vector<Effect::EffectBase*>, Effect::EffectBase*, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> >, Effect::EffectBase*>::variant_type {aka class boost::optional<boost::variant<std::vector<Effect::EffectBase*>, Effect::EffectBase*, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_, boost::detail::variant::void_> >}’
         typedef typename variant_type::types types;
</pre>


As well as

<pre>
In file included from /var/tmp/portage/games-strategy/freeorion-0.4.5/work/freeorion-0.4.5/util/SerializeUniverse.cpp:8:0:
/usr/include/boost/archive/detail/iserializer.hpp: In instantiation of ‘static void boost::archive::detail::heap_allocation<T>::doesnt_have_new_operator::invoke_delete(T*) [with T = Ship]’:
/usr/include/boost/archive/detail/iserializer.hpp:265:33:   required from ‘static void boost::archive::detail::heap_allocation<T>::invoke_delete(T*) [with T = Ship]’
/usr/include/boost/archive/detail/iserializer.hpp:273:26:   required from ‘boost::archive::detail::heap_allocation<T>::~heap_allocation() [with T = Ship]’
/usr/include/boost/archive/detail/iserializer.hpp:294:36:   required from ‘void* boost::archive::detail::pointer_iserializer<Archive, T>::heap_allocation() const [with Archive = boost::archive::binary_iarchive; T = Ship]’
/var/tmp/portage/games-strategy/freeorion-0.4.5/work/freeorion-0.4.5/util/SerializeUniverse.cpp:326:133:   required from here
/var/tmp/portage/games-strategy/freeorion-0.4.5/work/freeorion-0.4.5/util/../universe/Ship.h:99:5: error: ‘virtual Ship::~Ship()’ is protected
     ~Ship() {}
     ^
</pre>

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -166,7 +166,7 @@ if(${Boost_VERSION} LESS "105600")
     )
 endif()
 
-if(${Boost_VERSION} EQUAL "106100")
+if(${Boost_VERSION} EQUAL "106100" OR ${Boost_VERSION} GREATER "106100")
     # with boost 1.61 some boost::optional internals were changed. However
     # boost::spirit relies on some API the old implementation provided.  This
     # define enables the usage of the old boost::optional implementation.

